src/pages/index.js

```diff
@@ -0,0 +1,293 @@
+
+  return cardElement;
+}
+
+// Helper function to set up image preview
+function setUpImagePreview(imageElement, data) {
+  imageElement.addEventListener("click", () => {
+    const modalImage = previewModal.querySelector(".modal__image");
+    const modalCaption = previewModal.querySelector(".modal__caption");
+
+    modalImage.src = data.link;
+    modalImage.alt = data.name;
+    modalCaption.textContent = data.name;
+
+    openModal(previewModal);
+  });
+}
+
+// Function for Like button handler
+function setUpLikeButton(likeButton, data) {
+  if (data.isLiked) likeButton.classList.add("card__like-btn_liked");
+
+  likeButton.addEventListener("click", () => {
+    const isLiked = likeButton.classList.contains("card__like-btn_liked");
+    api
+      .handleLikeStatus(data._id, isLiked)
+      .then((updatedData) =>
+        likeButton.classList.toggle("card__like-btn_liked", updatedData.isLiked)
+      )
+      .catch(console.error);
+  });
+}
+
+// Function for Delete modal logic
+function openDeleteModal(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+
+  const deleteButton = deleteForm.querySelector(".modal__submit-btn");
+  showLoading(deleteButton, false, "Deleting..", "Delete"); // Reset button text to "Delete"
+
+  // Define a new delete handler for the delete action
+  const newDeleteHandler = (evt) => {
+    evt.preventDefault();
+    showLoading(deleteButton, true, "Deleting...", "Delete"); // Show "Deleting..." while processing
+
+    api
+      .deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove(); // Remove the selected card from DOM upon successful deletion
+        closeModal(deleteModal);
+      })
+      .catch((error) => console.error("Failed to delete card:", error))
+      .finally(() => showLoading(deleteButton, false, "Deleting", "Delete")); // Reset button text to "Delete"
+  };
+
+  // Remove any previously attached handler from deleteForm if it exists
+  deleteForm.removeEventListener("submit", deleteForm._deleteHandler);
+
+  // Store the new handler in a property and attach it to the delete form
+  deleteForm._deleteHandler = newDeleteHandler;
+  deleteForm.addEventListener("submit", newDeleteHandler, { once: true });
+
+  // Open the delete modal
+  openModal(deleteModal);
+}
+
+// Profile edit form submission
+editProfileForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  const saveButton = editProfileForm.querySelector(".modal__submit-btn");
+  showLoading(saveButton, true, "Saving...", "Save");
+
+  const name = editProfileForm.querySelector("#profile-name-input").value;
+  const about = editProfileForm.querySelector(
+    "#profile-description-input"
+  ).value;
+
+  api
+    .editUserInfo({ name, about })
+    .then((userData) => {
+      document.querySelector(".profile__name").textContent = userData.name;
+      document.querySelector(".profile__description").textContent =
```

Comment:
* It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine